### ISSUE TITLE
Bug: [prefer-find] Add support for checking ternaries

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#8379]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://main--typescript-eslint.netlify.app/rules/prefer-find

### Description

This addresses a minor omission when I did the PR for this rule (https://github.com/typescript-eslint/typescript-eslint/pull/8216).

I forgot to add support for checking ternary branches. I've made a quick branch that would add support. See https://github.com/typescript-eslint/typescript-eslint/commit/aa43ec626a718867568f028e0e148f894d3da739

### Fail

```TypeScript
(Math.random < 0.5 ? [1, 2, 3].filter(x => x % 2 === 0) : [4, 5, 6].filter(x => x % 2 === 0))[0]
```

### Pass

```TypeScript
Math.random < 0.5 ? [1, 2, 3].find(x => x % 2 === 0) : [4, 5, 6].find(x => x % 2 === 0)
```

**Comment by JoshuaKGoldberg** [comment]
Yeah makes sense, good catch. I'd even consider this a bug - there's no reason why the rule wouldn't also want to catch these.

If we were to consider this a feature, it would maybe arguably be a breaking change. But let's get this in quick as a bug. 😄
